.github/workflows/windows-hyperv-periodic.yml

```diff
@@ -0,0 +1,299 @@
+          ssh -i $HOME/.ssh/id_rsa ${{ env.SSH_OPTS }} azureuser@${{ env.VM_PUB_IP }} "sh.exe -s" << EOF
+            cd /c/containerd
+            export EXTRA_TESTFLAGS="-timeout=20m"
+            make integration | tee /c/Logs/integration.log
+          EOF
+          ssh -i $HOME/.ssh/id_rsa ${{ env.SSH_OPTS }} azureuser@${{ env.VM_PUB_IP }} "sh.exe -c 'cat /c/Logs/integration.log | go-junit-report.exe > /c/Logs/junit_00.xml'"
+
+      - name: PrepareRepoList
+        run: |
+            cat > containerd-hyperv-config.toml << EOF
+            version = 2
+
+            [plugins."io.containerd.grpc.v1.cri".containerd]
+            default_runtime_name = "runhcs-wcow-hypervisor"
+
+            [plugins."io.containerd.grpc.v1.cri".containerd.runtimes]
+                [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runhcs-wcow-hypervisor]
+                    base_runtime_spec = ""
+                    cni_conf_dir = ""
+                    cni_max_conf_num = 0
+                    container_annotations = []
+                    pod_annotations = []
+                    privileged_without_host_devices = false
+                    runtime_engine = ""
+                    runtime_path = ""
+                    runtime_root = ""
+                    runtime_type = "io.containerd.runhcs.v1"
+                    [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runhcs-wcow-hypervisor.options]
+                      Debug = true
+                      DebugType = 2
+                      SandboxImage = "mcr.microsoft.com/windows/servercore:1809"
+                      SandboxPlatform = "windows/amd64"
+                      SandboxIsolation = 1
+            EOF
+
+            cat > repolist.toml << EOF
+            busybox = "${{ env.BUSYBOX_TESTING_IMAGE_REF }}"
+            ResourceConsumer = "${{ env.RESOURCE_CONSUMER_TESTING_IMAGE_REF }}"
+            EOF
+
+            cat > cri-test-images.yaml << EOF
+            defaultTestContainerImage: ${{ env.BUSYBOX_TESTING_IMAGE_REF }}
+            webServerTestImage: ${{ env.WEBSERVER_TESTING_IMAGE_REF }}
+            EOF
+
+            scp -i $HOME/.ssh/id_rsa ${{ env.SSH_OPTS }} repolist.toml azureuser@${{ env.VM_PUB_IP }}:c:/repolist.toml
+            scp -i $HOME/.ssh/id_rsa ${{ env.SSH_OPTS }} cri-test-images.yaml azureuser@${{ env.VM_PUB_IP }}:c:/cri-test-images.yaml
+            scp -i $HOME/.ssh/id_rsa ${{ env.SSH_OPTS }} containerd-hyperv-config.toml azureuser@${{ env.VM_PUB_IP }}:c:/containerd-hyperv-config.toml
+
+      # NOTE(aznashwan): in-tree integration tests will need some updates to on
+      # Hyper-V containers so we skip this for now:
+      - name: RunCRIIntegrationTests
+        run: |
+           ssh -i $HOME/.ssh/id_rsa ${{ env.SSH_OPTS }} azureuser@${{ env.VM_PUB_IP }} "sh.exe -s" <<EOF
+             cd c:/containerd
+             ./script/setup/install-cni-windows
+             export TEST_IMAGE_LIST=c:/repolist.toml
+             export USE_HYPERV=1
```

Comment:
We also need to set ```WINDOWS_SANDBOX_IMAGE``` to the version of the operating system we're running this on. See above comment about ```SANDBOX_IMAGE```. 

This will hopefully not be needed in the future, and we'll be able to use 1809 images on ltsc2022. Until then, see discussions here:

https://github.com/containerd/containerd/pull/6491
and here:
https://github.com/containerd/containerd/issues/6508